proc/process.c

```diff
@@ -930,6 +1000,8 @@ static void process_exec(thread_t *current, process_spawn_t *spawn)
 	if (!err) {
 		stack = process_putargs(stack, &spawn->envp, &count);
 		stack = process_putargs(stack, &spawn->argv, &count);
+		/* align stack down */
+		stack = (void *)(((unsigned int)stack - 8) & ~7);
```

Comment:
```suggestion
		stack = (void *)(((ptr_t)stack - 8) & ~7);
```

Reply:
Won't this alignment backfire? We'll lose the position of our arguments, if we really move the stack here.

Something for later: We should preallocate the space on stack (we need to find out size of arguments) and put arguments from aligned place (bottom -> up) instead of putting them top -> down and hoping for aligned result (not is scope of this PR ofc)

Reply:
Refactored `hal_jmp` a little bit so that we won't end up with unaligned stack (and deleted this change)